plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php

```diff
@@ -954,8 +956,8 @@ private function has_block_checkout() {
 global $pagenow;
 if (
 	'customize.php' === $pagenow ||
-	isset( $_REQUEST['customize_theme'] ) || // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-	! wc_current_theme_is_fse_theme()
```

Comment:
@gigitux I wonder that moving this check to `WC_Shop_Customizer` constructor prevents calling `wc_current_theme_is_fse_theme()` early.

From my observation, not always, as we still call `wc_current_theme_is_fse_theme()` right away when initializing `WC_Shop_Customizer`.